### PR TITLE
xpar: init at 0.5

### PR DESCRIPTION
**Pull request by mrbenjadmin**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/xp/xpar/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    meta = {
+      description = "An error/erasure code system guarding data integrity.";
+      homepage = "https://github.com/kspalaiologos/xpar";
+      changelog = "https://github.com/kspalaiologos/xpar/blob/${version}/NEWS";
+      license = lib.licenses.gpl3Plus;
+      maintainers = with lib.maintainers; [ mrbenjadmin ];
+      platforms = lib.platforms.unix;
+    };
+  }
```
Comment (FliegendeWurst) [inline pkgs/by-name/xp/xpar/package.nix:40]: There should be another newline here

File: pkgs/by-name/xp/xpar/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ] ++ lib.optionals stdenv.hostPlatform.isx86_64 [ nasm ];
+
+    configureFlags = [
+      "--disable-arch-native"
+      "--enable-lto"
+    ] ++ lib.optional stdenv.hostPlatform.isx86_64 "--enable-x86-64"
+      ++ lib.optional stdenv.hostPlatform.isAarch64 "--enable-aarch64";
+
+    meta = {
+      description = "An error/erasure code system guarding data integrity.";
```
Comment (FliegendeWurst) [inline pkgs/by-name/xp/xpar/package.nix:33]:
```suggestion
      description = "Error/erasure code system guarding data integrity";
```

File: pkgs/by-name/xp/xpar/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      homepage = "https://github.com/kspalaiologos/xpar";
+      changelog = "https://github.com/kspalaiologos/xpar/blob/${version}/NEWS";
+      license = lib.licenses.gpl3Plus;
+      maintainers = with lib.maintainers; [ mrbenjadmin ];
+      platforms = lib.platforms.unix;
```
Comment (FliegendeWurst) [inline pkgs/by-name/xp/xpar/package.nix:38]:
```suggestion
      platforms = lib.platforms.all;
      mainProgram = "xpar";
```
It seems to compile for Windows too.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/xp/xpar/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+stdenv.mkDerivation {
+  pname = "xpar";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "kspalaiologos";
+    repo = "xpar";
+    rev = version;
+    hash = "sha256-DMUDWQqYSQjGxYOpcfwNaaM21avcZ1w3IqEhuOaabrw=";
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/xp/xpar/package.nix:21]: Better:
```suggestion

stdenv.mkDerivation (finalAttrs: {
  pname = "xpar";
  version = "0.5";

  src = fetchFromGitHub {
    owner = "kspalaiologos";
    repo = "xpar";
    rev = finalAttrs.version;
    hash = "sha256-DMUDWQqYSQjGxYOpcfwNaaM21avcZ1w3IqEhuOaabrw=";
  };
```

**Review by Aleksanaa — APPROVED** [review]
Looks good. Please squash commits into `maintainers: add mrbenjadmin` and `xpar: init at 0.5`

File: maintainers/maintainer-list.nix
Comment (FliegendeWurst) [inline maintainers/maintainer-list.nix:1]: Please drop the extra changes here

Comment (mrbenjadmin) [reply maintainers/maintainer-list.nix:1]: I apologise, that was an embarassing mistake! Thank you for your patience.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build xpar

**Pull request merged**